packages/core/src/components/side-pane-group/side-pane-group.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Siemens AG
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import {
+  Component,
+  Element,
+  h,
+  Host,
+  Listen,
+  Prop,
+  State,
+} from '@stencil/core';
+
+const mobileBreakpoint: number = 767;
+
+@Component({
+  tag: 'ix-side-pane-group',
+  styleUrl: 'side-pane-group.scss',
+  shadow: true,
+})
+export class SidePaneGroup {
+  @Element() hostElement: HTMLIxSidePaneGroupElement;
+
+  /**
+   * Determines if the side panes behave inline
+   */
+  @Prop() inline: boolean = false;
+
+  /**
+   * Determines if the side panes behave floating
+   */
+  @Prop() floating: boolean = false;
+
+  /**
+   * Choose the variant of the panes
+   */
+  @Prop() variant: '1' | '2' = '2';
+
+  @State() mobileMode: boolean = false;
+
+  componentWillRender() {
+    if (this.inline && this.floating) {
+      console.error('Inline and floating can not be set at once!');
+      return;
+    }
+
+    this.onWindowResize();
+  }
+
+  componentDidRender() {
+    const sidePanels = this.getSidePanels();
+    sidePanels.forEach((sidePanelElement) => {
+      sidePanelElement.inline = this.inline;
+      sidePanelElement.floating = this.floating;
+    });
+
+    this.configureLayout();
+  }
+
+  configureLayout() {
+    const sidePanels = this.getSidePanels();
+    sidePanels.forEach((sidePanelElement) => {
+      let zIndex: string = '1';
```

Comment:
On the typescript code we use the type-inference feature. (https://www.typescriptlang.org/docs/handbook/type-inference.html). Type definition can be removed here